File: 2015/day01/pt1.py
# takes direction list, returns specified floor
def translator(directions):
    flr = 0
    for i in range(len(directions)):
        if directions[i] == '(':
            flr += 1
        if directions[i] == ')':
            flr -= 1
    return flr


# takes a document and copies it to a list
def copier(doc):
    directions = []
    for i in doc:
        directions.append(doc[i])
    return directions


# main program?
file = 'input.txt'      # This is all you have to change to deal with dif. input file names
with open(file) as fp:
    directions = fp.readline()
    print(directions)
    properFloor = translator(directions)
    print(properFloor)

File: 2015/day01/pt1.c
#include <stdio.h>
#define MAXCHAR 9999999



// takes directions array of arbitrary length and returns the specified floor
// can loose the counter processing and just pass in size after measure mod is working
int translator ( char directions[] ) {
   int i, floor, counter;
   
   floor = 0;

   for ( i=0; i<1000; i++ ) {
      if ( directions[i] != '\0' ) { counter = counter + 1; }
      else break;
   }

   for ( i=0; i < counter; i++ ) {
      if ( directions[i] == '(' ) { floor = floor + 1; }
      if ( directions[i] == ')' ) { floor = floor - 1; }
   }

   return floor;
}



// main program
int main ( void ) {
   FILE* fp;
   char *buffer;

   buffer = (char *) malloc( MAXCHAR * sizeof( char ) );  

   fp = fopen( "input.txt", "r" );

   while ( fgets( buffer, MAXCHAR, fp ) ) {
      

   printf ( "ta-da: %d \n", translator( temp ) );

   return 0;
}
